src/Service/StatisticsExtractionService.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/**
+ * @file Contains service that extracts statistics that should be delivered to
+ * Faktor.
+ */
+
+namespace App\Service;
+
+use App\Document\Entry;
+use App\Document\ExtractionResult;
+use App\Repository\EntryRepository;
+use App\Repository\ExtractionResultRepository;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class StatisticsExtractorService.
+ */
+class StatisticsExtractionService
+{
+    protected const BATCH_SIZE = 50;
+
+    private $documentManager;
+    private $logger;
+    private $extractionResultRepository;
+    private $elasticsearchService;
+    private $entryRepository;
+
+    /**
+     * StatisticsExtractionService constructor.
+     *
+     * @param \Doctrine\ODM\MongoDB\DocumentManager $documentManager
+     *   The document manager
+     * @param \App\Repository\EntryRepository $entryRepository
+     *   Repository for Entry documents
+     * @param \App\Repository\ExtractionResultRepository $extractionResultRepository
+     *   Repository for ExtractResult documents
+     * @param \Psr\Log\LoggerInterface $logger
+     *   The logger
+     * @param \App\Service\ElasticsearchServiceInterface $elasticsearchService
+     *   Service to integrate with elasticsearch
+     */
+    public function __construct(DocumentManager $documentManager, EntryRepository $entryRepository, ExtractionResultRepository $extractionResultRepository, LoggerInterface $logger, ElasticsearchServiceInterface $elasticsearchService)
+    {
+        $this->documentManager = $documentManager;
+        $this->logger = $logger;
+        $this->elasticsearchService = $elasticsearchService;
+        $this->extractionResultRepository = $extractionResultRepository;
+        $this->entryRepository = $entryRepository;
+    }
+
+    /**
+     * Extract new statistics.
+     *
+     * @throws \Doctrine\ODM\MongoDB\MongoDBException
+     * @throws \Throwable
+     */
+    public function extractStatistics()
+    {
+        $today = new \DateTime();
+
+        // Get latest extraction entry. Default to first day of production.
+        /* @var ExtractionResult $lastExtraction */
+        $lastExtraction = $this->extractionResultRepository->getNewestEntry();
+
+        // Default to 1. december 2019 to make sure we extract all statistics from the start of production.
```

Comment:
We might need to add here that it's the production date for http://cover.dandigbib.org/ and not the faktor service.

Reply:
Fixed